### ISSUE TITLE
winrm SSL error on Ansible 2.2.0.0 with pywinrm<0.2.1

### ISSUE THREAD

**Issue by ghost** [ansible/ansible#17990]
##### ISSUE TYPE
- Documentation Report
##### COMPONENT NAME

connections to Windows devices
##### ANSIBLE VERSION

2.2.0.0
##### CONFIGURATION

N/A
##### OS / ENVIRONMENT

Windows Server 2012 R2
##### SUMMARY

When connecting to windows server 2012 (potentially others, unknown) devices with Ansible 2.1.2.0 and pywinrm 0.2.0, there are no issues.

When upgrading to Ansible 2.2.0.0 without upgrading pywinrm, the following occurs:

```
<SNIP>
"/var/lib/awx/venv/ansible/lib/python2.7/site-packages/OpenSSL/_util.py", line
48, in exception_from_error_queue     raise exception_type(errors) Error:
[('SSL routines', 'SSL3_WRITE_PENDING', 'bad write retry')]

<IP Address> WINRM CLOSE SHELL: GUID
fatal: [Hostname]: FAILED! => {
    "failed": true, 
    "msg": "winrm send_input failed"
}
```

If the full stacktrace is necessary or useful I can provide it.

The docs currently recommend pywinrm>=0.1.1 Here: https://github.com/ansible/ansible/blob/devel/docsite/rst/intro_windows.rst

which is potentially valid up to 2.1.2.0, but no longer appears to be so.

I know that pywinrm is a soft dependency because it's only necessary for windows, but it's likely something that will impact others given the new windows modules that come in with 2.2, so I thought I would bring it up.  I also was not sure what the policy/process was for updating the docs w/r/t which version(s) the intro docs target, which is why I didn't just put in a PR with a simple version bump.

Cheers.

**Comment by nitzmahone** [comment]
Check that your requests version actually meets the requirement (>=2.10.0)- you didn't specify, but I'd bet you're running on Ubuntu. There's a default-installed requests 2.2 there that pip will silently fail to upgrade (even though pywinrm has declared 2.10.0 as the minimum required)- HTTPS with any requests version older than 2.10 can be subject to this issue (it's a well-known intermittent failure way down deep in urllib3/pyopenssl and has nothing to do with the Ansible version). 

Check with `pip list`, and if it's less than 2.10.0, you can usually still get pip to do the right thing with `pip install requests>=2.10.0 --upgrade`.

**Comment by ghost** [comment]
Thanks for looking into this.

I do not think that was the issue, though.

I have run hundreds, if not thousands of jobs on Tower with Ansible 2.1.2.0 without ever seeing this error, and then every single job failed against all Windows targets when I updated Ansible to 2.2.0.0, until I also updated the version of pywinrm to 0.2.1.  I did some amount of testing involving upgrading and then rolling back individual pip and system packages one at a time until the problem was resolved, though I did not test all packages - I stopped when it started working again.

For what it's worth though:

```
root@prod-tower:~# /var/lib/awx/venv/ansible/bin/pip list | grep requests[\ ]
requests (2.11.1)
```

**Comment by nitzmahone** [comment]
pywinrm 0.2.1 is a red herring too then, because it only included a one-line exception handler fix for Kerberos delegation over 0.2.0.

**Comment by nitzmahone** [comment]
Ansible 2.2.0 also made zero changes to the winrm connection plugin and its interface with pywinrm over 2.1. There's _something_ else that got changed on the fly.

**Comment by nitzmahone** [comment]
This is almost certainly an OS packaging issue (which is outside Ansible's hands, since we have no control over OS packaging)- if you're seeing this error, you're using an old broken version of requests that does not match pywinrm's setup.py requirements. It's been fixed since requests 2.10, which is why that's our minimum version.
